govuk_bank_holidays/bank_holidays.py

```diff
@@ -184,8 +192,60 @@ def get_prev_work_day(self, division=None, date=None):
         """
         date = date or datetime.date.today()
         one_day = datetime.timedelta(days=1)
-        holidays = set(holiday['date'] for holiday in self.get_holidays(division=division))
+        holidays = self.get_known_holiday_date_set(division)
         while True:
             date -= one_day
             if date.weekday() not in self.weekend and date not in holidays:
                 return date
+
+    def holidays_after(self, division=None, date=None):
+        """
+        Yields known bank holidays in chronological order
+        NB: If no division is specified, only holidays common to *all* divisions are yielded.
+        :param division: see division constants; defaults to common holidays
+        :param date: starting after this date; defaults to today
+        """
+        date = date or datetime.date.today()
+        holidays = self.get_holidays(division=division)
+        yield from filter(lambda holiday: holiday['date'] > date, holidays)
+
+    def holidays_before(self, division=None, date=None):
+        """
+        Yields known bank holidays in reverse chronological order
+        NB: If no division is specified, only holidays common to *all* divisions are yielded.
+        :param division: see division constants; defaults to common holidays
+        :param date: starting before this date; defaults to today
+        """
+        date = date or datetime.date.today()
+        holidays = reversed(self.get_holidays(division=division))
+        yield from filter(lambda holiday: holiday['date'] < date, holidays)
+
+    def work_days_after(self, division=None, date=None):
+        """
+        Yields an infinite series of work days in chronological order skipping weekends and known bank holidays
+        NB: If no division is specified, only holidays common to *all* divisions are yielded.
+        :param division: see division constants; defaults to common holidays
+        :param date: starting after this date; defaults to today
+        """
+        date = date or datetime.date.today()
+        one_day = datetime.timedelta(days=1)
+        holidays = self.get_known_holiday_date_set(division)
+        while True:
+            date += one_day
+            if date.weekday() not in self.weekend and date not in holidays:
```

Comment:
I notice this expression quite few times and AFAICT this could be replaced with `self.is_work_day(date, division)` which should be equivalent but reads much better.

Reply:
lemme see if i can refactor…

Reply:
yes, you're quite right.